index.bs

```diff
@@ -1370,77 +896,768 @@ What configuration is the device in after it resets?
+    </td>
+  </tr>
+  <tr>
+    <td><dfn>\[[selectedAlternateSetting]]</dfn></td>
+    <td>An <a>empty</a> list of <a>integer</a></td>
+    <td>
+      The current alternate setting for each interface on the current configuration.
+    </td>
+  </tr>
+  <tr>
+    <td><dfn>\[[claimedInterface]]</dfn></td>
+    <td>An <a>empty</a> list of <a>boolean</a></td>
+    <td>
+      The claimed status for each interface on the current configuration.
+    </td>
+  </tr>
+</table>
+
+<div algorithm="find the device descriptor for the connected USB device">
+    To <dfn>find the device descriptor for the connected USB device</dfn>,
+    perform the following steps:
+    1. Let |deviceDescriptor| be the <a>device descriptor</a> of the connected
+        device by performing <a>Get Descriptor</a> with
+        <code>DescriptorType</code> set to <code>DEVICE</code>.
+    1. Return |deviceDescriptor|.
+</div>
+
+<div algorithm="find a list of configuration descriptors for the connected USB device">
+    To <dfn>find a list of configuration descriptors for the connected USB device</dfn>,
+    perform the following steps:
+    1. Let |deviceDescriptor| be the result of
+        <a>finding the device descriptor for the connected USB device</a>.
+    1. Let |numConfigurations| be <code>bNumConfigurations</code> of |deviceDescriptor|.
+    1. Let |configurationIndex| be 0.
+    1. Let |configurationDescriptors| be an <a>empty</a> <a>list</a>.
+    1. While |configurationIndex| is less than |numConfigurations|:
+        1. Let |descriptors| be the a list of <a>descriptors</a>
+            by performing <a>Get Descriptor</a> with
+            <code>DescriptorType</code> set to <code>CONFIGURATION</code> and
+            <code>DescriptorIndex</code> set to |configurationIndex|.
+        1. If the <code>bDescriptorType</code> of the |descriptors|[0] is equal to <code>CONFIGURATION</code>,
+            <a>append</a> |descriptors|[0] to |configurationDescriptors|.
+        1. Increment |configurationIndex| by 1.
+    1. Return |configurationDescriptors|.
+</div>
+
+<div algorithm="abort transfers currently scheduled on an interface">
+    To <dfn>abort transfers currently scheduled on an interface</dfn> with the given |interface| object,
+    perform the following steps:
+    1. Let |configuration| be |interface|.{{USBInterface/[[configuration]]}}.
+    1. Let |device| be |configuration|.{{USBConfiguration/[[device]]}}.
+    1. If |configuration| is not the same as the result of <a>finding the current configuration</a>
+        with |device|, return.
+    1. If the result of <a>finding if the interface is claimed</a> with |interface|
+        is not <code>true</code>, return.
+    1. Let |currAlternateInterface| be the result of
+        <a>finding the alternateInterface for the current alternate setting</a> with |interface|.
+    1. <a>For each</a> |endpoint| of |currAlternateInterface|.{{USBAlternateInterface/endpoints}}:
```

Comment:
```suggestion
    1. <a>For each</a> |endpoint| of |currAlternateInterface|.{{USBAlternateInterface/[[endpoints]]}}:
```

Reply:
Updated in https://github.com/WICG/webusb/pull/212/commits/5aaf696938ed1dc073f8cc8da8c809e9e5019f41.